core/lib/types/src/tx/execute.rs

```diff
@@ -72,7 +72,10 @@ impl EIP712TypedStructure for Execute {
     const TYPE_NAME: &'static str = "Transaction";
 
     fn build_structure<BUILDER: StructBuilder>(&self, builder: &mut BUILDER) {
-        builder.add_member("to", &U256::from(self.contract_address.as_bytes()));
+        builder.add_member(
+            "to",
+            &U256::from(self.contract_address.unwrap_or_default().as_bytes()),
+        );
```

Comment:
is it correct way to do it? AFAIU it means that signatures for txs with `tx.contract_address = null` and `tx. contract_address = Address::zero()` will be the same which doesn't look correct

Reply:
I agree that it might be incorrect for the signature to remain the same when the address is null or `0x00`. I've changed it to only include the `to` field as a member when the contract address is `Some(..)`, and not include it when it's `None`. 

I'm not entirely sure if this is the best approach or if it could introduce additional issues. While reviewing other parts of the code, I noticed that when a member has an `Option` type, it usually defaults when it is `None`, like [here](https://github.com/matter-labs/zksync-era/blob/3f1d1a26076d5bf2fd65c31876d2a32a3cbfff0c/core/lib/types/src/transaction_request.rs#L339), but this might be a case where it's not possible to get `None` as a value. Let me know it there's a better way to do this